parsers/numword.go

```diff
@@ -0,0 +1,94 @@
+// Given a number in string form
+// Return the number in word form of the greatest power (e.g. 300,100,000,000 => 300.1 Billion)
+// Lower limit > Thousands (e.g. xxx,000) # Numbers below this are in hundreds and can be expressed numerically
+// Upper limit > Centillion (10^303) ref: https://en.wikipedia.org/wiki/Names_of_large_numbers#Standard_dictionary_numbers
+
+package parsers
+
+import (
+  "math"
+  "regexp"
+  "strings"
+  "strconv"
+)
+
+// NumberWord handles strings made of contiguous "0-9" characters
+// converts to word strings of the greatest power
+type NumberWord struct{}
+
+// NewNumberWord construcs a NumberWord struct
+func NewNumberWord() *NumberWord {
+  return &NumberWord{}
+}
+
+// CanParseIntoHuman ...
+func (n *NumberWord) CanParseIntoHuman(s string) bool {
+  // is it 4 or more characters? (e.g. is it => 1000)
+  if len(s) < 4 {
+    return false
+  }
+
+  // is it a (delimited[,. ]) number?
+  match, _ := regexp.MatchString(`^(([0-9]+)|([0-9]{1,3}[., ])+[0-9]{1,3})$`, s)
+  if match {
+    // is it less than the max? 
+      //  300000 max places plus delimiters
+    if len(s) < 400000 {
+      return true
+    }
+  }
+  // everything else is not a number
+  return false
```

Comment:
me gusta !!!

The only other thing I would do (and this is a personal style preference so take it with a grain of salt lol) is move the inline comments into the "docs comments" above the function declaration on line 24

ie:

```golang
// CanParseIntoHuman determines if string is within the bounds the parser works in; in that
// the input needs to be more than 4 characters, delimited by commas, no more
// 300_000 etc etc etc
func () ...
```

This way when we run `go doc` on the code to make the document website, we'll capture some of the requirements for input and expose them to readers of said website.